src/main/java/org/nekobehemoth/hw06/executor/CustomExecutorService.java

```diff
@@ -0,0 +1,201 @@
+package org.nekobehemoth.hw06.executor;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class CustomExecutorService implements ExecutorService {
+
+    BlockingQueue<Runnable> queue;
+    private final Set<Worker> workers = new HashSet<>();
+    private final AtomicInteger activeTasks;
+    private final AtomicInteger completedTasks;
+    private volatile boolean isShutdown;
+    private volatile boolean terminated = false;
+    private final boolean useVirtualThreads;
+    private final ReentrantLock lock = new ReentrantLock();
+
+    private static final Runnable SHUTDOWN = () -> {};
+
+    public CustomExecutorService(int corePoolSize, boolean useVirtualThreads) {
+        this.useVirtualThreads = useVirtualThreads;
+        this.activeTasks = new AtomicInteger();
+        this.completedTasks = new AtomicInteger();
+        this.queue = new LinkedBlockingQueue<>();
+        ThreadFactory threadFactory = useVirtualThreads ? Thread.ofVirtual().factory() : Thread.ofPlatform().factory();
+        for (int i = 0; i < corePoolSize; i++) {
+            Worker worker = new Worker(threadFactory);
+            this.workers.add(worker);
+            worker.getThread().start();
+        }
```

Comment:
difficult to test, violates SRP, difficult to use with a DI container. 
if a service has a lifecycle it's better to initialize it separately